### PR TITLE
Create table component

### PR DESCRIPTION
**Pull request by patjanawat**
Create a table component and sub table (Collapse & Expand)

Related issue https://github.com/mesg-foundation/mesg-components/issues/6

### CODE REVIEW

File: components/Table/Table.vue [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            <div :style="align(head.align)">{{ head.text}}</div>
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="(row, rowIndex) in items" :key="rowIndex">
+          <td v-for="(header,i) in headers" :key="i">
+            <div :style="align(header.align)">
+              <a v-if="header.link" :href="row[header.value]" target="_blank">
+                <img :src="src(header.icon)" :srcset="srcset(header.icon)" />
```
Comment (antho1404) [inline components/Table/Table.vue]: why is there an image for every link? this component should be generic.

File: components/Table/Table.vue [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+              </a>
+              <div v-else-if="header.collapse" class="collapse">
+                <img :src="src(header.icon)" :srcset="srcset(header.icon)" />
```
Comment (antho1404) [inline components/Table/Table.vue]: same here and I also don't understand what there is this part

File: components/Table/Table.vue [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "mesgTable",
```
Comment (antho1404) [inline components/Table/Table.vue]: Let's keep the same than on the other components

```suggestion
  name: "Table",
```

**Review by antho1404 — CHANGES_REQUESTED** [review]
Why are there images on this PR?
Also please have a look at how to create a table component in vue and probably the concept of "slot"

File: components/Table/Table.vue (hunk 3 above)
Comment (patjanawat) [reply components/Table/Table.vue]: Done.

File: components/Table/Table.vue (hunk 1 above)
Comment (patjanawat) [reply components/Table/Table.vue]: I will change this like slot. Thanks.

File: components/Table/Table.vue (hunk 2 above)
Comment (patjanawat) [reply components/Table/Table.vue]: Line 16, display link image and line 18 display expand and collapse icon (down & up)

File: components/TableRow/TableRow.vue [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+          </div>
+        </slot>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "TableRow",
```
Comment (antho1404) [inline components/TableRow/TableRow.vue]: This component is not a table and row and it's also not generic at all so this should be done at the application level, not on the library... this component should be removed

File: components/Table/Table.vue [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+              <td v-show="showExpand">                
+                <i class="fas" :class="src(item.id)"></i>
+              </td>
+            </slot>
+          </tr>
+          <tr v-show="showExpand" v-if="toggle.includes(item.id)" :key="item.id">
+            <td :colspan="columnSpan">
+              <slot name="expandItem" :distributions="item.distributions"> 
+                <!-- default content -->
+                <div class="item-background"> Hello world </div>
```
Comment (antho1404) [inline components/Table/Table.vue]: why a hello world in a generic component?

File: components/Table/Table.vue [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+              <!-- default content -->
+              <td v-for="(header,i) in headers" :key="i">{{ item[header.value]}}</td>
+              <td v-show="showExpand">                
+                <i class="fas" :class="src(item.id)"></i>
+              </td>
+            </slot>
+          </tr>
+          <tr v-show="showExpand" v-if="toggle.includes(item.id)" :key="item.id">
+            <td :colspan="columnSpan">
+              <slot name="expandItem" :distributions="item.distributions">
```
Comment (antho1404) [inline components/Table/Table.vue]: this component doesn't need to know about distribution what if I want to use it for a user list for example.

File: components/Table/Table.vue [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+<script>
+export default {
+  name: "Table",
+  components: {    
+  },
+  props: {
+    headers: Array,
+    items: Array,
+    showExpand: Boolean,
+    subTitle:String
```
Comment (antho1404) [inline components/Table/Table.vue]: this is never used

File: components/Table/Table.vue [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  },
+  data() {
+    return {
+      expanded: [],
+      isExpand: false,
+      decimalDigit: 4,
```
Comment (antho1404) [inline components/Table/Table.vue]: this is never used

File: plugin/filter.date.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import Vue from 'vue'
```
Comment (antho1404) [inline plugin/filter.date.js]: what the purpose of that filter? where this is used?

File: plugin/filter.number.js [hunk 10]
```diff
@@ -0,0 +1,4 @@
+import Vue from 'vue'
```
Comment (antho1404) [inline plugin/filter.number.js]: same here why having this?

**Comment by antho1404** [comment]
images are still in the pull request and there is no need for such images. This has to be removed

File: components/TableRow/TableRow.vue (hunk 4 above)
Comment (patjanawat) [reply components/TableRow/TableRow.vue]: Removed.

File: components/Table/Table.vue (hunk 7 above)
Comment (patjanawat) [reply components/Table/Table.vue]: headers use for a display column name and items use for display a row data.
showExpand use for expand and collapse row detail and removed subTitle prop.

File: components/Table/Table.vue (hunk 8 above)
Comment (patjanawat) [reply components/Table/Table.vue]: removed.

File: plugin/filter.date.js (hunk 9 above)
Comment (patjanawat) [reply plugin/filter.date.js]: Plugin out to date. Removed

File: plugin/filter.number.js (hunk 10 above)
Comment (patjanawat) [reply plugin/filter.number.js]: Plugin out to date. Removed

File: components/Table/Table.vue (hunk 5 above)
Comment (patjanawat) [reply components/Table/Table.vue]: Default content.

File: components/Table/Table.vue (hunk 6 above)
Comment (patjanawat) [reply components/Table/Table.vue]: Removed distribution. If you want to use it you can create user list content and into the slot.

**Pull request closed without merging**